pkg/controller/volume/persistentvolume/pv_controller_base.go

```diff
@@ -243,6 +243,10 @@ func (ctrl *PersistentVolumeController) deleteClaim(claim *v1.PersistentVolumeCl
 	_ = ctrl.claims.Delete(claim)
 	glog.V(4).Infof("claim %q deleted", claimToClaimKey(claim))
 
+	if claim.Spec.VolumeName == "" {
```

Comment:
Please move this check below `volumeName := claim.Spec.VolumeName` and use `volumeName` in the check

Reply:
Consider logging here as well.